File: 2515. Shortest Distance to Target String in a Circular Array/solution.cpp
class Solution {
public:
  int closetTarget(vector<string>& words, string target, int startIndex) {
    int shortestDistance = -1;
    for (int index = 0; index < words.size(); index++){
      string word = words[index];
      if (word == target){
        int distance = abs(index - startIndex);
        int distance2 = words.size() - abs(index - startIndex);
        if (shortestDistance > distance || shortestDistance == -1){
          shortestDistance = distance;
        }
        if (shortestDistance > distance2){
          shortestDistance = distance2;
        }
      }
    }

    return shortestDistance;
  }
};

File: 2515. Shortest Distance to Target String in a Circular Array/solution.py
class Solution(object):
  def closetTarget(self, words, target, startIndex):
    """
    :type words: List[str]
    :type target: str
    :type startIndex: int
    :rtype: int
    """
    shortestDistance = -1
    for index, word in enumerate(words):
      if word == target:
        distance = abs(index - startIndex)
        distance2 = len(words) - abs(index - startIndex)
        if shortestDistance > distance or shortestDistance == -1:
          shortestDistance = distance
        if shortestDistance > distance2:
          shortestDistance = distance2

    return shortestDistance